mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -1,12 +1,22 @@
 package scala.meta.internal.semver
 
+import scala.util.Try
+
 object SemVer {
 
-  case class Version(major: Int, minor: Int, patch: Int) {
+  case class Version(
+      major: Int,
+      minor: Int,
+      patch: Int,
+      releaseCandidate: Int,
+      milestone: Int
+  ) {
     def >(that: Version): Boolean = {
       this.major > that.major ||
       (this.major == that.major && this.minor > that.minor) ||
-      (this.major == that.major && this.minor == that.minor && this.patch > that.patch)
+      (this.major == that.major && this.minor == that.minor && this.patch > that.patch) ||
+      (this.major == that.major && this.minor == that.minor && this.patch == that.patch && this.releaseCandidate > that.releaseCandidate) ||
+      (this.major == that.major && this.minor == that.minor && this.patch == that.patch && this.releaseCandidate == that.releaseCandidate && this.milestone > that.milestone)
```

Comment:
As you notice, we have to compare the Pre-Release version in a different way.

FYI, here's the logic of comparing semantic versions in scalafmt
https://github.com/scalameta/scalafmt/blob/f05136cdd5ceeeabdb6054658d315f7a0a3d025e/scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtVersion.scala#L15-L18


Following the logic from scalafmt, these two lines would be something like this (if we use `0` for no pre-release version):

```scala
(this.major == that.major && this.minor == that.minor && this.patch == that.patch && isLaterPreRelease(this.releaseCandidate, that.releaseCandidate)) ||
(this.major == that.major && this.minor == that.minor && this.patch == that.patch && isLaterPreRelease(this.releaseCandidate, that.releaseCandidate)) && isLaterPreRelease(this.milestone, that.milestone)

// returns true if v1 is later than v2 (for pre-release version)
def isLaterPreRelease(v1: Int, v2: Int): Boolean {
  v1 == 0 || (v2 !== 0 && v1 > v2)
}
```

Comparing two semantic versions is always complexed 😅

Reply:
Thanks for the hint! I went with the Option solution suggested by @ckipp01